pkg/retagger/job.go

```diff
@@ -20,6 +21,9 @@ type JobOptions struct {
 	TagSuffix string
 
 	OverrideRepoName string
+
+	// UpdateOnChange sets whether a pattern Job should update the destination image if a source image changes for a given tag
```

Comment:
I think we would _nearly_ never want to do this. We have retagger because we want our images to be stable and to guarantee that they don't move. The option to move a tag should therefor never be included.

WDYT?

Reply:
The main use I anticipate would be for tags like `latest` or `stable`, for example.

Reply:
Are there cases where we would want to retag `latest` or `stable`? I think we should always work against immutable tags here. We generally don't want to follow moving tags to keep everything reproducible / testable and so on.

Reply:
I don't like moving tags, but I see two reasons we might want it. The first is for automatically testing images which publish `dev` or `latest` or whatever but not tags for pre-release versions (so not hyperkube, but smaller stuff). Like "can we run this latest app version?" We can also just agree to never do this.

The bigger thing is that we _should_ be informed if a tag does actually change upstream because if an otherwise "fixed" tag gets moved it is probably either a big mistake or a very serious reason.

I agree we always want to use fixed version numbers everywhere else

Reply:
So as it turns out, the image ID reported by Quay is not the actual image ID, making the check for changed digests actually useless. If we want to check for changed tags, we will either need to pull every image again (obviously not worth it) or use some secondary storage where we keep the SHA we pushed to Quay at acquisition and compare it on subsequent checks. That actually isn't so bad, but I'm not sure if it's worth it. So in light of that, I think it is ok to remove the Update feature